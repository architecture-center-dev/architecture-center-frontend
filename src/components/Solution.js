import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import { red } from '@material-ui/core/colors';
import FavoriteIcon from '@material-ui/icons/Favorite';

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: 345,
  },
  media: {
    height: 0,
    paddingTop: '56.25%', // 16:9
  },
  expand: {
    transform: 'rotate(0deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
  avatar: {
    backgroundColor: red[500],
  },
}));

export default function RecipeReviewCard() {
  const classes = useStyles();
  
  return (
    <Card className={classes.root}>
      <CardHeader
        avatar={
          <Avatar aria-label="recipe" className={classes.avatar}>
            CK
          </Avatar>
        }
        
        title="E-commerce checkout"
        subheader="December 29, 2020"
      />
      <CardMedia
        className={classes.media}
        image="/checkout.jpg"
        title="Paella dish"
      />
      <CardContent>
        <Typography variant="body2" color="textSecondary" component="p">
            Checkout solution to be used via API or embedding on the page via javascript
        </Typography>
      </CardContent>
      <CardActions disableSpacing>
        <IconButton aria-label="add to favorites">
          <FavoriteIcon />
        </IconButton>
      </CardActions>
      
    </Card>
  );
}
