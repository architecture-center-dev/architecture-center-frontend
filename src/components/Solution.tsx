import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Typography from '@material-ui/core/Typography';
import Link from "next/link"

const useStyles = makeStyles(() => ({
  root: {
    maxWidth: 345,
    minWidth: 345
  },
  media: {
    height: 0,
    paddingTop: '56.25%', // 16:9
  },
}));

export type SolutionProps = {
  solution_id: string,
  name: string,
  description: string,
  date: string,
  image: string
}

export default function Solution({ solution_id, name, description, date, image }: SolutionProps) {
  const classes = useStyles();

  return (
    <Card className={classes.root} data-testid="solution">
      <Link href={`/solution/${solution_id}`}>
        <CardHeader
          title={name}
          subheader={date}
        />
      </Link>
      <CardMedia
        className={classes.media}
        image={image}
        title={name}
      />
      <CardContent>
        <Typography variant="body2" color="textSecondary" component="p">
          {description}
        </Typography>
      </CardContent>
      <CardActions disableSpacing>

      </CardActions>

    </Card>
  );
}