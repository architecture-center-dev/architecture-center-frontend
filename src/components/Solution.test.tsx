import * as React from "react";
import Solution, { SolutionProps } from "./Solution";
import { render } from '@testing-library/react'

describe("<Solution />", () => {
  it.only("Component should render correctely", async () => {

    const props: SolutionProps = {
      solution_id: "solution_id",
      name: "ame",
      description: "description",
      date: "ate",
      image: "image"
    }

    const { getByTestId, getByText } = render(<Solution {...props} />);
    const solution = getByTestId('solution');

    expect(solution).toBeTruthy();
    expect(getByText(props.name)).toBeInTheDocument;
    expect(getByText(props.description)).toBeInTheDocument;
    expect(getByText(props.date)).toBeInTheDocument;
    expect(solution).toMatchSnapshot();
  });
});
