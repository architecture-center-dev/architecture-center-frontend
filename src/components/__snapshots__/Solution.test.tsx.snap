// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Solution /> Component should render correctely 1`] = `
<div
  class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="solution"
>
  <div
    class="MuiCardHeader-root"
  >
    <div
      class="MuiCardHeader-content"
    >
      <span
        class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
      >
        ame
      </span>
      <span
        class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        ate
      </span>
    </div>
  </div>
  <div
    class="MuiCardMedia-root makeStyles-media-2"
    style="background-image: url(image);"
    title="ame"
  />
  <div
    class="MuiCardContent-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      description
    </p>
  </div>
  <div
    class="MuiCardActions-root"
  />
</div>
`;
